pkg/render/kube-controllers.go

```diff
@@ -423,6 +428,34 @@ func (c *kubeControllersComponent) controllersDeployment() *apps.Deployment {
 	return &d
 }
 
+// prometheusService creates a Service which exposes and endpoint on kube-controllers for
+// reporting Prometheus metrics.
+func (c *kubeControllersComponent) prometheusService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{Kind: "Service", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "calico-kube-controllers-metrics",
+			Namespace: common.CalicoNamespace,
+			Labels:    map[string]string{"k8s-app": "calico-kube-controllers"},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{"k8s-app": "calico-kube-controllers"},
+			Type:     v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				// Expose on the default port - 9094.
+				// TODO: Users can change this value in KubeControllersConfiguration. We should
```

Comment:
Wouldn't it be relatively easy to watch KubeControllersConfiguration and read this field. Is there a good reason to not do that yet?